src/yb/util/memory/arena.h

```diff
@@ -284,6 +287,11 @@ class ArenaBase {
   // allocated and added to the arena's list. Also covers any other
   // mutation of the component data structure (eg Reset).
   mutable mutex_type component_lock_;
+
+  MemTrackerPtr mem_tracker_ = nullptr;
+  void TrackMemory(size_t bytes);
+  void UntrackMemory(size_t bytes);
+  void TrackMemoryDifference(size_t bytes_before, size_t bytes_after);
```

Comment:
I just realized that `MemTracker::Consume` can also take a negative integer as input, so in this case, we don't need this member function anymore.